mcp/sensors/star_camera_receive.c

```diff
@@ -0,0 +1,347 @@
+        sc2_hm_Addr = channels_find_by_name("sc2_hm");
+        sc2_focus_position_Addr = channels_find_by_name("sc2_focus_position");
+        sc2_focus_inf_Addr = channels_find_by_name("sc2_focus_inf");
+        sc2_aperture_steps_Addr = channels_find_by_name("sc2_aperture_steps");
+        sc2_max_aperture_Addr = channels_find_by_name("sc2_max_aperture");
+        sc2_min_focus_pos_Addr = channels_find_by_name("sc2_min_focus_pos");
+        sc2_max_focus_pos_Addr = channels_find_by_name("sc2_max_focus_pos");
+        sc2_current_aperture_Addr = channels_find_by_name("sc2_current_aperture");
+        sc2_exposure_time_Addr = channels_find_by_name("sc2_exposure_time");
+        sc2_focus_mode_Addr = channels_find_by_name("sc2_focus_mode");
+        sc2_start_focus_pos_Addr = channels_find_by_name("sc2_start_focus_pos");
+        sc2_end_focus_pos_Addr = channels_find_by_name("sc2_end_focus_pos");
+        sc2_focus_step_Addr = channels_find_by_name("sc2_focus_step");
+        sc2_photos_per_focus_Addr = channels_find_by_name("sc2_photos_per_focus");
+    }
+    SET_SCALED_VALUE(sc2_spike_limit_Addr, scr.blobParams[0]);
+    SET_SCALED_VALUE(sc2_dyanmic_hp_Addr, scr.blobParams[1]);
+    SET_SCALED_VALUE(sc2_r_smooth_Addr, scr.blobParams[2]);
+    SET_SCALED_VALUE(sc2_high_pass_filter_Addr, scr.blobParams[3]);
+    SET_SCALED_VALUE(sc2_r_high_pass_filter_Addr, scr.blobParams[4]);
+    SET_SCALED_VALUE(sc2_search_border_Addr, scr.blobParams[5]);
+    SET_SCALED_VALUE(sc2_filter_return_Addr, scr.blobParams[6]);
+    SET_SCALED_VALUE(sc2_n_sigma_Addr, scr.blobParams[7]);
+    SET_SCALED_VALUE(sc2_unique_star_spacing_Addr, scr.blobParams[8]);
+    SET_SCALED_VALUE(sc2_make_static_hp_mask_Addr, scr.makeHP);
+    SET_SCALED_VALUE(sc2_use_static_hp_mask_Addr, scr.useHP);
+    SET_SCALED_VALUE(sc2_time_limit_Addr, scr.solveTimeLimit);
+    SET_SCALED_VALUE(sc2_logodds_Addr, scr.logOdds);
+    SET_SCALED_VALUE(sc2_latitude_Addr, scr.latitude);
+    SET_SCALED_VALUE(sc2_longitude_Addr, scr.longitude);
+    SET_SCALED_VALUE(sc2_hm_Addr, scr.heightWGS84);
+    SET_SCALED_VALUE(sc2_focus_position_Addr, scr.focusPos);
+    SET_SCALED_VALUE(sc2_focus_inf_Addr, scr.setFocusInf);
+    SET_SCALED_VALUE(sc2_aperture_steps_Addr, scr.apertureSteps);
+    SET_SCALED_VALUE(sc2_max_aperture_Addr, scr.maxAperture);
+    SET_SCALED_VALUE(sc2_min_focus_pos_Addr, scr.minFocusPos);
+    SET_SCALED_VALUE(sc2_max_focus_pos_Addr, scr.maxFocusPos);
+    SET_SCALED_VALUE(sc2_current_aperture_Addr, scr.aperture);
+    SET_SCALED_VALUE(sc2_exposure_time_Addr, scr.exposureTime);
+    SET_SCALED_VALUE(sc2_focus_mode_Addr, scr.focusMode);
+    SET_SCALED_VALUE(sc2_start_focus_pos_Addr, scr.startPos);
+    SET_SCALED_VALUE(sc2_end_focus_pos_Addr, scr.endPos);
+    SET_SCALED_VALUE(sc2_focus_step_Addr, scr.focusStep);
+    SET_SCALED_VALUE(sc2_photos_per_focus_Addr, scr.photosPerStep);
+}
+
+static void where_to_unpack(struct star_cam_return data, int which_sc) {
```

Comment:
Better function name needed. What does this function do? Doesn't it actually DO the assignment of sc data to one or the other set of telemetry addresses?

Reply:
Needs function comment with doxygen style

Reply:
Both done.